# HG changeset patch
# User John W. Eaton <jwe@octave.org>
# Date 1518296776 18000
#      Sat Feb 10 16:06:16 2018 -0500
# Node ID af11cc7cd33639f6081f04ce1fb6e0bc36ed2d8e
# Parent  11e359eb63f4476caff7a5d9efa05339b867e00f
use DEFMETHODX instead of undefining quad

* quad.cc: Don't #undef quad.
(Fquad): Use DEFMETHODX.

* mk-builtins.pl: Don't emit code to undefine quad.

diff --git a/libinterp/corefcn/quad.cc b/libinterp/corefcn/quad.cc
--- a/libinterp/corefcn/quad.cc
+++ b/libinterp/corefcn/quad.cc
@@ -40,20 +40,16 @@ along with Octave; see the file COPYING.
 #include "ovl.h"
 #include "ov-fcn.h"
 #include "unwind-prot.h"
 #include "utils.h"
 #include "variables.h"
 
 #include "Quad-opts.cc"
 
-#if defined (quad)
-#  undef quad
-#endif
-
 // Global pointer for user defined function required by quadrature functions.
 static octave_function *quad_fcn;
 
 // Have we warned about imaginary values returned from user function?
 static bool warned_imaginary = false;
 
 // Is this a recursive call?
 static int call_depth = 0;
@@ -125,18 +121,18 @@ quad_float_user_function (float x)
         }
 
       retval = tmp(0).xfloat_value ("quad: expecting user supplied function to return numeric value");
     }
 
   return retval;
 }
 
-DEFMETHOD (quad, interp, args, ,
-           doc: /* -*- texinfo -*-
+DEFMETHODX ("quad", Fquad, interp, args, ,
+            doc: /* -*- texinfo -*-
 @deftypefn  {} {@var{q} =} quad (@var{f}, @var{a}, @var{b})
 @deftypefnx {} {@var{q} =} quad (@var{f}, @var{a}, @var{b}, @var{tol})
 @deftypefnx {} {@var{q} =} quad (@var{f}, @var{a}, @var{b}, @var{tol}, @var{sing})
 @deftypefnx {} {[@var{q}, @var{ier}, @var{nfun}, @var{err}] =} quad (@dots{})
 Numerically evaluate the integral of @var{f} from @var{a} to @var{b} using
 Fortran routines from @w{@sc{quadpack}}.
 
 @var{f} is a function handle, inline function, or a string containing the
diff --git a/libinterp/mk-builtins.pl b/libinterp/mk-builtins.pl
--- a/libinterp/mk-builtins.pl
+++ b/libinterp/mk-builtins.pl
@@ -149,20 +149,16 @@ elsif ($make_source)
 #endif
 
 #include \"builtin-defun-decls.h\"
 #include \"help.h\"
 #include \"ovl.h\"
 #include \"symtab.h\"
 #include \"variables.h\"
 
-#if defined (quad)
-#  undef quad
-#endif
-
 namespace octave
 {";
 
   @installer_functions = ();
 
   foreach $arg (@ARGV)
   {
     $file = $arg;
